### PR TITLE
[fix](replica) handle replica version missing info to avoid -214 error

### LINKED ISSUE #8208: [Bug] query failed with -214
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

0.15

### What's Wrong?

Some time query may failed with error code: -214

```
error Error 5025: errCode = 2, detailMessage = failed to initialize storage reader. tablet=9929137.1435609437.e64a4f69d3271f54-98005e0571265fb7, res=-214, backend=192.168.1.1
```

-214 is error code from BE side: `OLAP_ERR_VERSION_NOT_EXIST`
which means there are missing version of tablet.

### What You Expected?

Although the replica with missing version will be repaired automatically, but it may takes a long time.
We should let FE know the replica's missing version info and do not choose that replica to do the query,
to avoid -214 error during the repair.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #8208 

## Problem Summary:

In the original tablet reporting information, the version missing information is done by combining
two pieces of information as follows:

1. the maximum consecutive version number
2. the `version_miss` field

The logic of this approach is confusing and inconsistent with the logic of checking for missing versions when querying.

After the change, we directly use the version checking logic used in the query, and set `version_miss` to true
if a missing version is found

and on the FE processing side. Originally, only the **bad replica** information was syncronized among FEs,
but not the **version missing** information. As a result, the non-master FE is not aware of the missing version information.

In the new design, we deprecate the original log persistence class `BackendTabletsInfo` and use the new 
`BackendReplicasInfo` to record replica reporting information and write both **bad** and **version missing**
information to metadata so that other FEs can synchronize these information.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -577,9 +578,9 @@ OLAPStatus Tablet::capture_consistent_versions(const Version& spec_version,
     return status;
 }
 
-OLAPStatus Tablet::check_version_integrity(const Version& version) {
+OLAPStatus Tablet::check_version_integrity(const Version& version, bool quite) {
```
Comment (yangzhg) [inline be/src/olap/tablet.cpp:581]: what does quite mean?

Comment (morningman) [reply be/src/olap/tablet.cpp:581]: if !quite, it will print log for missing versions.

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by yangzhg — DISMISSED** [review]
LGTM

File: be/src/olap/tablet.cpp (hunk 1 above)
Comment (yangzhg) [reply be/src/olap/tablet.cpp:581]: You mean quiet? slient may be better

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**